File: lib/src/core/models/runtime_mode.dart
enum RuntimeMode {
  vpn,
  localProxy,
}

extension RuntimeModeLabel on RuntimeMode {
  String get label {
    switch (this) {
      case RuntimeMode.vpn:
        return 'VPN 模式';
      case RuntimeMode.localProxy:
        return '本地代理';
    }
  }
}

File: native/Sources/XrayNativeCore/Models/RuntimeMode.swift
import Foundation

public enum RuntimeMode: String, Codable, CaseIterable, Sendable, Identifiable {
    case vpn
    case localProxy

    public var id: String { rawValue }

    public var label: String {
        switch self {
        case .vpn:
            return "VPN 模式"
        case .localProxy:
            return "本地代理"
        }
    }
}
